### PR TITLE
[ADD] w_pos_order: List of POS order on website

### PR DESCRIPTION
**Pull request by remytms**
This add the list of POS order for a user in the website.

This has not been fully tested yet.

### CODE REVIEW

File: beesdoo_website_pos_order/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2015-2016 Odoo S.A.
+# Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
```
Comment (houssine78) [inline beesdoo_website_pos_order/controllers/main.py:4]: remove non contributor to this code

File: beesdoo_website_pos_order/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+from openerp import http
+from openerp.exceptions import AccessError, MissingError
+from openerp.http import request
+
+from openerp.addons.website_portal_v10.controllers.main import WebsiteAccount
+
+
+class CooperatorWebsiteAccount(WebsiteAccount):
+
```
Comment (houssine78) [inline beesdoo_website_pos_order/controllers/main.py:19]: replace by PosOrderWebsiteAccount(WebsiteAccount):

File: beesdoo_website_pos_order/controllers/main.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    @http.route()
+    def account(self):
+        """ Add POS Order to main account page """
+        response = super(CooperatorWebsiteAccount, self).account()
+        partner = request.env.user.partner_id
+
+        pos_order_mgr = request.env['pos.order'].sudo()
```
Comment (houssine78) [inline beesdoo_website_pos_order/controllers/main.py:26]: sudo() should not come here. always instantiate it on the function call. here you put a sudo() wildcard on all the function calls which is maybe not recommended

File: beesdoo_website_pos_order/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        partner = request.env.user.partner_id
+        pos_order_mgr = request.env['pos.order'].sudo()
+        pos_order = pos_order_mgr.browse(oid)
+        try:
+            if pos_order.partner_id != partner.commercial_partner_id:
+                raise Forbidden()
+        except AccessError:
+            raise Forbidden()
+        except MissingError:
+            raise NotFound()
```
Comment (houssine78) [inline beesdoo_website_pos_order/controllers/main.py:95]: how these raises are handled in the client side? what will see the customer if an error is raised?

File: beesdoo_website_pos_order/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    def get_pos_order_receipt(self, oid=-1):
+        """Render the receipt for a POS order"""
+        # Get the POS order and raise an error if the user
+        # is not allowed to access to it or if the object is not found.
+        partner = request.env.user.partner_id
+        pos_order_mgr = request.env['pos.order'].sudo()
+        pos_order = pos_order_mgr.browse(oid)
+        try:
+            if pos_order.partner_id != partner.commercial_partner_id:
+                raise Forbidden()
```
Comment (houssine78) [inline beesdoo_website_pos_order/controllers/main.py:91]: according the domain it should not happen.

File: beesdoo_website_pos_order/controllers/main.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        except AccessError:
+            raise Forbidden()
+        except MissingError:
+            raise NotFound()
+        # Get the pdf
+        report_mgr = request.env['report'].sudo()
```
Comment (houssine78) [inline beesdoo_website_pos_order/controllers/main.py:97]: sudo() should be moved to the function call insted of here(cfr my remark on the other PR)

File: beesdoo_website_pos_order/views/beesdoo_website_pos_order_templates.xml [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+     Copyright 2015-2016 Odoo S.A.
+     Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+     Copyright 2018 Rémy Taymans <remytaymans@gmail.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+-->
+<openerp>
+
+  <!-- POS Order in the menu -->
+  <template
+      id="portal_my_home_menu_pos_order"
```
Comment (houssine78) [inline beesdoo_website_pos_order/views/beesdoo_website_pos_order_templates.xml:12]: remove my_home. portal_menu_pos_order is enough

**Review by houssine78 — CHANGES_REQUESTED** [review]

File: beesdoo_website_pos_order/controllers/main.py (hunk 4 above)
Comment (remytms) [reply beesdoo_website_pos_order/controllers/main.py:95]: It's ok, this generate proper 404 NotFound page or 403 Forbidden page in the right design. :)

**Comment by remytms** [comment]
This PR is old. And the functionality is not needed anymore.

**Pull request closed without merging**